<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Graph Algorithms</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="loadGraphButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>120</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Load Graph</string>
    </property>
   </widget>
   <widget class="QPushButton" name="saveGraphButton">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>20</y>
      <width>120</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Save Graph</string>
    </property>
   </widget>
   <widget class="QComboBox" name="algorithmComboBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>70</y>
      <width>260</width>
      <height>30</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Breadth-First Search</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Depth-First Search</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Connected Components</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Build Hasse Diagram</string>
     </property>
    </item>
   </widget>
   <widget class="QLineEdit" name="startVertexLineEdit">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>70</y>
      <width>50</width>
      <height>30</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Start Vertex</string>
    </property>
   </widget>
   <widget class="QPushButton" name="runAlgorithmButton">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>70</y>
      <width>120</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Run Algorithm</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="resultsTextEdit">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>120</y>
      <width>540</width>
      <height>220</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="timeLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>360</y>
      <width>200</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Time: 0 ms</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar"/>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
